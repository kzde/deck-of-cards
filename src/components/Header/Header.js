import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import './Header.scss';

const Header = (props) => {

    const shuffleHandler = () => props.handleShuffle();
    const dealOneCardHandler = () => props.handleDealOneCard();

    return (
        <div className="Header">
            <AppBar position="static">
                <Typography variant="h6" color="inherit" >
                    Deck Of Cards
                </Typography>
                <Button color="inherit" onClick={shuffleHandler}>Shuffle</Button>
                <Button color="inherit" onClick={dealOneCardHandler}>Deal One Card</Button>
            </AppBar>
        </div>
    )
}

Header.propTypes = {
    handleShuffle: PropTypes.func.isRequired,
    handleDealOneCard: PropTypes.func.isRequired
}

export default Header;