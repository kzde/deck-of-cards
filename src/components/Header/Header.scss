.Header {
    header{
        display: flex;
        flex-direction: row;
        align-items: center;

        h6 {
            margin: 10px 20px;
        }
    }
}