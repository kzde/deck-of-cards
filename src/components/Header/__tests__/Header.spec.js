import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';
jest.mock('@material-ui/core/AppBar', () => 'AppBar');
jest.mock('@material-ui/core/Button', () => 'Button');
jest.mock('@material-ui/core/Typography', () => 'Typography');

const props = {
    handleShuffle: () => { },
    handleDealOneCard: () => { }
}
describe('Header', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(<Header {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})