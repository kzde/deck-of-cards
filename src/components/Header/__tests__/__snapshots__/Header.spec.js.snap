// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders correctly 1`] = `
<div
  className="Header"
>
  <AppBar
    position="static"
  >
    <Typography
      color="inherit"
      variant="h6"
    >
      Deck Of Cards
    </Typography>
    <Button
      color="inherit"
      onClick={[Function]}
    >
      Shuffle
    </Button>
    <Button
      color="inherit"
      onClick={[Function]}
    >
      Deal One Card
    </Button>
  </AppBar>
</div>
`;
